tokio-trace/src/lib.rs

```diff
@@ -49,153 +49,7 @@
 //! # }
```

Comment:
I think if we're not going to give any more stuff here on spans, it'd be good to make this example a little more expansive. I would also avoid using the word "context" here, because as far as I've gathered, it's entirely up to the subscriber whether or not work done below `.enter()` is considered "in the context of" the span. The span _only_ serves to build a pair of start and end events, and does not say anything about events "contained" within that.

Reply:
> I think if we're not going to give any more stuff here on spans, it'd be good to make this example a little more expansive

I thought the docs would be more readable if more of the content in the lib-level docs was pulled out into module-level docs, so the reader isn't immediately presented with a massive wall of text. The hope was that the lib docs would serve as an overview/table of contents, and most of the actual documentation would live in the appropriate module. I'd love to know if you think this is actually a good way to structure things, or if it's not working for you?

That said, I can definitely add a more complete example here. I wasn't sure about including an example that shows a lot of concepts that haven't been introduced yet in the documentation, though?

> I would also avoid using the word "context" here, because as far as I've gathered, it's entirely up to the subscriber whether or not work done below .enter() is considered "in the context of" the span. The span only serves to build a pair of start and end events, and does not say anything about events "contained" within that.

I think there is an expectation that there's some _concept_ of "context", but the policy of what a "context" means is up to the subscriber. I'll try and edit these docs to make it clearer that this is provided by the subscriber implementation.

Reply:
Oh, sorry, let me rephrase: I think it's great to move things out of the top-level docs the way you have. I meant more that given that you're going to do that, the examples that are left behind should probably be _slightly_ more elaborate than the examples that were there before. I don't think it's too problematic to have more complex concepts in the example here, since the user will know that to get a detailed explanation they should go to the module-level docs.